baselines/hfedxgboost/README.md

```diff
@@ -0,0 +1,278 @@
+dataset: [a9a, cod-rna, ijcnn1, space_ga] # list of datasets you include in your baseline
+---
+
+# HFedXgboost: Gradient-less Federated Gradient Boosting Trees with Learnable Learning Rates
+
+> Note: If you use this baseline in your work, please remember to cite the original authors of the paper as well as the Flower paper.
+
+****Paper:**** https://arxiv.org/abs/2304.07537
+****Authors:**** Chenyang Ma, Xinchi Qiu, Daniel J. Beutel, Nicholas D. Lane
+
+****Abstract:**** The privacy-sensitive nature of decentralized datasets and
+the robustness of eXtreme Gradient Boosting (XGBoost) on
+tabular data raise the need to train XGBoost in the con-
+text of federated learning (FL). Existing works on federated
+XGBoost in the horizontal setting rely on the sharing of gradients, which induce per-node level communication frequency
+and serious privacy concerns. To alleviate these problems, we
+develop an innovative framework for horizontal federated
+XGBoost which does not depend on the sharing of gradients and simultaneously boosts privacy and communication
+efficiency by making the learning rates of the aggregated
+tree ensembles are learnable. We conduct extensive evaluations
+on various classification and regression datasets, showing
+our approach achieves performance comparable to the state-of-the-art method and effectively improves communication
+efficiency by lowering both communication rounds and communication overhead by factors ranging from 25x to 700x.
+
+
+## About this baseline
+
+****What’s implemented:**** The code in this directory replicates the experiments in "Gradient-less Federated Gradient Boosting Trees with Learnable Learning Rates" (Ma et al., 2023) for a9a, cod-rna, ijcnn1, space_ga datasets, which proposed the FedXGBllr algorithm. Concretely, it replicates the results for a9a, cod-rna, ijcnn1, space_ga datasets in Table 2.
+
+****Datasets:**** a9a, cod-rna, ijcnn1, space_ga
+
+****Hardware Setup:**** 
+
+****Contributors:**** [Aml Hassan Esmil](https://github.com/Aml-Hassan-Abd-El-hamid)
+
+## Experimental Setup
+
+****Task:**** Tabular classification and regression
+
+****Model:****: XGBoost model combined with 1-layer CNN
+
+****Dataset:**** 
+This baseline only includes 7 datasets with a focus on 4 of them (a9a, cod-rna, ijcnn1, space_ga).
+
+Each dataset can be partitioned across 2, 5 or 10 clients in an IID distribution.
+
+| Dataset | task type | no.of features | no.of samples | 
+| :---: | :---: | :---: | :---: |
+| a9a | Binary classification | 123 | 32,561 |
+| cod-rna | Binary classification | 8 | 59,5358 |
+| ijcnn1 | Binary classification | 22 | 49,990 |
+| abalone | regression | 8 | 4,177 |
+| a9a | regression | 123 | 32,561 |
+| cpusmall | regression | 12 | 8,192 |
+| space_ga | regression | 6 | 3,167 |
+| YearPredictionMSD | regression | 90 | 515,345 |
+
+
+****Training Hyperparameters:**** 
```

Comment:
Maybe there is no need to list them all, only the key ones. I think because you found it critical to have different hyperparameters for each dataset and client configuration, you needed to have quite a few `.yaml` files in `conf/clients`. This is fine.  But i think briefly mentioning this would be good to know right from the beginning for someone interested in running your baseline.